### PR TITLE
Convert cachekey plugin to use Regex class (PCRE2) instead of PCRE

### PR DESCRIPTION
**Pull request by bryancall**
## Summary
This PR converts the cachekey plugin from using the old PCRE library directly to using the modern `ts::Regex` class which wraps PCRE2.

## Changes
- Replace direct PCRE usage with `ts::Regex` class
- Update Pattern class to use `Regex::compile()` and `Regex::exec()`
- Use `RegexMatches` for capture group handling
- Use `RE_ERROR_NOMATCH` from `tsutil/Regex.h`
- Add comprehensive error checking in `match()`, `capture()`, and `replace()` methods
- Use default parameter values (cleaner code)

## Benefits
- **Modern PCRE2 API**: Better performance and features
- **JIT compilation**: Automatic optimization
- **Better error handling**: Detailed error messages with error codes
- **RAII resource management**: No manual memory management needed
- **Thread-safe execution**: Safe concurrent regex operations

## Testing
- **Comprehensive test suite**: 141 assertions across 4 test cases (up from 33)
- **New test coverage**:
  - Config string parsing (`/pattern/replacement/` format)
  - Multiple replacement patterns and edge cases
  - `process()` method for both capture and replace modes
  - `MultiPattern` class with priority matching
  - `NonMatchingMultiPattern` exclusion logic
  - `Classifier` for User-Agent device detection
- **Backward compatibility verified**: All tests pass on both old PCRE and new PCRE2 implementations
- All existing functionality preserved

## Files Changed
- `plugins/cachekey/pattern.h`: Updated to use `Regex` class
- `plugins/cachekey/pattern.cc`: Converted from PCRE to PCRE2/Regex
- `plugins/cachekey/unit_tests/pattern_test.cc`: Added comprehensive tests

## Related
- Depends on #12606 (Expose PCRE2 error codes in tsutils Regex.h API)

### CODE REVIEW

File: plugins/cachekey/pattern.h [hunk 1]
```diff
@@ -18,29 +18,23 @@
...
  */
 class Pattern
 {
 public:
-  static const int TOKENCOUNT = 10;             /**< @brief Capturing groups $0..$9 */
-  static const int OVECOUNT   = TOKENCOUNT * 3; /**< @brief pcre_exec() array count, handle 10 capture groups */
+  static const int TOKENCOUNT = 10; /**< @brief Capturing groups $0..$9 */
 
   Pattern();
   virtual ~Pattern();
```
Comment (bneradt) [inline plugins/cachekey/pattern.h]: I don't think there's inheritance of Pattern, right? We should be able to remove `virtual` and set this ` = default;` here. And remove `Pattern::~Pattern() {}` in the .cc file.

File: plugins/cachekey/pattern.cc [hunk 2]
```diff
@@ -299,18 +270,11 @@ Pattern::replace(const String &subject, String &result)
...
-    int replIndex = _tokens[i];
-    int start     = ovector[2 * replIndex];
-    int length    = ovector[2 * replIndex + 1] - ovector[2 * replIndex];
-
-    /* Handle the case when no match / a group capture result in an empty string */
-    if (start < 0) {
-      start  = 0;
-      length = 0;
-    }
+    int              replIndex = _tokens[i];
```
[the remark this answers is not in the document]
Comment (bryancall) [reply plugins/cachekey/pattern.cc:268]: this check is done above on [line 264](https://github.com/apache/trafficserver/pull/12607/files#diff-0f9197c7533c4751f2b03aaad1bc1ec792376a3cbf54914f97612928d921419dR264)

**Comment by bryancall** [comment]
## Addressed Review Comments

### ✅ Fixed: Destructor (pattern.h line 40)
Removed `virtual` keyword from destructor and changed to `= default` as suggested by @bneradt. Pattern class has no inheritance, so the virtual destructor was unnecessary.

**Changes:**
- Header: Changed `virtual ~Pattern();` to `~Pattern() = default;`
- Implementation: Removed empty destructor from pattern.cc

### ℹ️ Copilot Comments - No Changes Needed

**Bounds Check (pattern.cc line 273):** This is a false positive. The bounds check already exists at lines 265-268 where all tokens are validated before use in the loop.

**String Construction (pattern.cc line 229):** The code is already consistent - both line 229 and line 277 use the same pattern: `String dst(capture.data(), capture.length());`
